src/actions/posts.js

```diff
@@ -364,6 +364,40 @@ export function editPost(post) {
     });
 }
 
+export function setUnreadPost(userId, postId) {
+    return async (dispatch, getState) => {
+        let unreadChan;
+        const state = getState();
+        try {
+            unreadChan = await Client4.markPostAsUnread(userId, postId);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+        let delta;
+        if (state.entities.channels.myMembers[unreadChan.channel_id]) {
```

Comment:
Since we're waiting on an async client call, the redux state may have changed since you called `getState`. You should move the call to happen after the `await`

Reply:
while I did that intentionally, I now think that you're right, if something changed the number of messages we probably want the new difference, I can't recall why I thought it was better to do it before the await. If I remember I'll let you know :)